### PR TITLE
Add scripts to run device tests

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

This PR adds the bits required to run the device tests - either from the terminal or on CI.

I am splitting each platform into separate cake files so that the arguments can be focused and easy to use. For example, the simplest case to run tests would be:

```
dotnet cake eng/devices/<platform>.cake --project=$PWD/xxx/xxx.csproj
```

For example:
```
dotnet cake eng/devices/ios.cake \
  --project=$PWD/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests.iOS/Xamarin.Platform.Handlers.DeviceTests.iOS.csproj

dotnet cake eng/devices/android.cake \
  --project=$PWD/src/Essentials/test/DeviceTests.Android/Xamarin.Essentials.DeviceTests.Android.csproj 
```

There is also a way just to run an output app file without building the project, then use the `--app` argument:

```
dotnet cake eng/devices/<platform>.cake --project=$PWD/xxx/xxx.csproj
```

For example:
```
dotnet cake eng/devices/ios.cake \
  --app=$PWDsrc/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests.iOS/bin/iPhoneSimulator/Release/XamarinEssentialsDeviceTestsiOS.app
```

There is either a way to do everything and build and run in one go, or you can decide:
 * `--target=test` (default) - build and test
 * `--target=build` - just build and not run
 * `--target=test --exclusive` - just test and not run

In addition to what to do, you can specify more accurate paths:
 * `--project=<path-to-csproj>` - the path to the platform .csproj file (required)
 * `--binlog=<path-to-binlog>` - the path to store the binlog file of the build (default: xxx.csproj.binlog)
 * `--app=<pth-to-apk-or-app>` - the path to the .app folder o the .apk file (default: the first app/apk in the bin directory)
 * `--results=<path-to-results-dir>` - the path to a directory that will hold all the test results and device logs (default: xxx.app-results)
 * `--device=<device-combo>` - the target ID of the device used by xharness in the form  
   `<platform>-<device|simulator|emulator>[-<32|64>][_<ios-version|android-api>]`  
   _eg: ios-simulator-64_13.4,android-emulator-32_30,[...]_
 * `--package=<android-package-name>` - the Android app package name found in the manifest
 * `--instrumentation=<android-instrumentation-name>` - the Android instrumentation name

### Issues Resolved ### 

- Fixes part of https://github.com/dotnet/maui/issues/329

### API Changes ###

 None

### Platforms Affected ### 

None

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

The cake files can just be run using either the default args or any set of valid, custom args. This should not require any other work - except maybe to install the emulators using the SDK manager. They are created, started and deleted after each run.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by mattleibow** [comment]
Tis needs to be added to the CI yaml, but that should probably be done in a separate PR.

**Comment by rmarinho** [comment]
I wonder if the build of the project  that we have in the other cake script should be used..

**Comment by mattleibow** [comment]
> I wonder if the build of the project that we have in the other cake script should be used..

That can be done too. You can pass in the app package instead of the project and then add the --exclusive arg. This will skip the build and run the tests directly. We could potentially have a bunch of jobs that pop out different variants of some app, and then just run all the tests on each app.

File: src/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests.Android/Xamarin.Platform.Handlers.DeviceTests.Android.csproj [hunk 1]
```diff
@@ -10,8 +10,8 @@
     <TemplateGuid>{122416d6-6b49-4ee2-a1e8-b825f31c79fe}</TemplateGuid>
     <OutputType>Library</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Xamarin.Platform.Handlers.DeviceTests.Android</RootNamespace>
-    <AssemblyName>Xamarin.Platform.Handlers.DeviceTests.Android</AssemblyName>
+    <RootNamespace>Maui.DeviceTests.Android</RootNamespace>
```
Comment (rmarinho) [inline src/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests.Android/Xamarin.Platform.Handlers.DeviceTests.Android.csproj:13]: If we are chaging this, should we just change the csproj naming also?

Comment (PureWeen) [reply src/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests.Android/Xamarin.Platform.Handlers.DeviceTests.Android.csproj:13]: No we'll do that once we're more set on a name

**Review by PureWeen — APPROVED** [review]

**Pull request merged**